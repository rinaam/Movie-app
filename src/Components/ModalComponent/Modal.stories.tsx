import React from "react";
import { Story, Meta } from "@storybook/react";

import ModalComponent, { IModalProps } from "./ModalComponent";

export default {
  title: "Modal",
  component: ModalComponent,
} as Meta;

const Template: Story<IModalProps> = (args) => <ModalComponent {...args} />;

export const Primary = Template.bind({});
Primary.args = {
  title: "Movie Roulette",
  open: true,
  children: (
    <div>
      <p>test</p>
      <p>llllll</p>
    </div>
  ),
};
