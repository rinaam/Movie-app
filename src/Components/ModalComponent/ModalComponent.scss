@use "sass:map";
@import "../../Styles/breakpoints.scss";
@import "../../Styles/variables.scss";

.modal {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background-color: $black-opacity;
  .container {
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    background-color: $white;
    width: 100%;
    height: 100%;
    padding: map.get($spacing, "s");
    display: flex;
    flex-direction: column;
    box-sizing: border-box;
    @include breakpoint(smallDevices) {
      width: 300px;
      height: 300px;
    }
    &-header {
      display: flex;
      justify-content: space-between;
      align-items: center;
      border-bottom: 1px solid $light-grey;
    }
    &-content {
      margin: map.get($spacing, "xs");
      height: 100%;
      flex: 1;
    }
  }
}
