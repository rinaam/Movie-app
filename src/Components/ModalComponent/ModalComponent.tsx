import React from "react";
import "./ModalComponent.scss";
import CloseIcon from "../../Assets/Icons/CloseIcon.svg";
import ButtonComponent from "../ButtonComponent/ButtonComponent";

export interface IModalProps {
  title: string;
  children: JSX.Element;
  open: boolean;
  onClose: () => void;
}

const ModalComponent: React.FC<IModalProps> = ({
  title,
  children,
  open,
  onClose,
}) => (
  <>
    {Boolean(open) && (
      <div className="modal">
        <div className="container">
          <div className="container-header">
            <p>{title}</p>
            <ButtonComponent variant="close" onClick={onClose}>
              <CloseIcon />
            </ButtonComponent>
          </div>
          <div className="container-content">{children}</div>
        </div>
      </div>
    )}
  </>
);

export default ModalComponent;
