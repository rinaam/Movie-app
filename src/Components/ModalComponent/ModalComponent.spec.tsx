import { create } from "react-test-renderer";
import ModalComponent from "./ModalComponent";
import React from "react";

describe("ModalComponent", () => {
  it("should render ModalComponent", () => {
    const fakeChildren = <p>Hello World</p>;
    const wrapper = create(
      <ModalComponent open title="Fake Title" onClose={jest.fn()}>
        {fakeChildren}
      </ModalComponent>
    ).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
