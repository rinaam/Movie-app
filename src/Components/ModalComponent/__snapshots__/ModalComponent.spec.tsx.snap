// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalComponent should render ModalComponent 1`] = `
<div
  className="modal"
>
  <div
    className="container"
  >
    <div
      className="container-header"
    >
      <p>
        Fake Title
      </p>
      <button
        className="btn btn-close"
        onClick={[MockFunction]}
        type="button"
      >
        <IconMock />
      </button>
    </div>
    <div
      className="container-content"
    >
      <p>
        Hello World
      </p>
    </div>
  </div>
</div>
`;
